src/commonComponents/sidebar/Sidebar.js

```diff
@@ -0,0 +1,84 @@
+// react
+import { useEffect, useState } from "react";
+
+// prop-types
+import PropTypes from "prop-types";
+
+//lodash
+import _map from "lodash/map";
+import _startCase from "lodash/startCase";
+
+// constants
+import { INITIAL_INDEX, SCROLL_FACTOR } from "./constants/sidebar.general";
+
+//helpers
+import { sanitizeText } from "../../utility/sanitizeText";
+import { getMatchedIndexWithUrlHash } from "./helpers/sidebar.general";
+
+// components
+import LinkItem from "../linkItem";
+
+// css
+import "./sidebar.css";
+
+const Sidebar = (props) => {
+  const { links, className } = props;
+
+  const [activeIdx, setActiveIdx] = useState(INITIAL_INDEX);
+
+  useEffect(() => {
+    const setInitialActiveIdx = () => {
+      const index = getMatchedIndexWithUrlHash(links);
+      setActiveIdx(index);
+    };
+
+    setInitialActiveIdx();
+  }, [links]);
+
+  const onClick = (index) => {
+    return function handleClick() {
+      setActiveIdx(index);
+    };
+  };
+
+  const renderLink = (label, index) => {
+    const sanitizedText = sanitizeText(label);
+
+    const isActive = activeIdx === index;
+
+    return (
+      <LinkItem
+        key={label}
+        label={_startCase(label)}
+        href={`#${sanitizedText}`}
+        isActive={isActive}
+        onClick={onClick(index)}
+      />
+    );
+  };
+
+  const scrollStyle = () => {
```

Comment:
another approach for scrolling that i am thinking is we can create a margin kind of thing for the active category and add transition so that it looks like scrolling

Reply:
but I think this is a better and simple solution